tests/testsuite/warning_override.rs

```diff
@@ -0,0 +1,170 @@
+//! Tests for overriding warning behavior using `build.warnings` config option.
+
+use std::sync::LazyLock;
+
+use cargo_test_support::{cargo_test, project, str, tools, Project};
+use snapbox::data::Inline;
+
+const ALLOW_CLEAN: LazyLock<Inline> = LazyLock::new(|| {
+    str![[r#"
+[CHECKING] foo v0.0.1 ([ROOT]/foo)
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]]
+});
+
+const ALLOW_CACHED: LazyLock<Inline> = LazyLock::new(|| {
+    str![[r#"
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]]
+});
+
+static WARN: LazyLock<Inline> = LazyLock::new(|| {
+    str![[r#"
+...
+[WARNING] unused variable: `x`
+...
+[WARNING] `foo` (bin "foo") generated 1 warning
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]]
+});
+
+const DENY: LazyLock<Inline> = LazyLock::new(|| {
+    str![[r#"
+...
+[WARNING] unused variable: `x`
+...
+[WARNING] `foo` (bin "foo") generated 1 warning
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[ERROR] warnings are denied by `build.warnings` configuration
+
+"#]]
+});
+
+fn make_project_with_rustc_warning() -> Project {
+    project()
+        .file(
+            "Cargo.toml",
+            &format!(
+                r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2021"
+            "#
+            ),
+        )
+        .file("src/main.rs", "fn main() { let x = 3; }")
+        .build()
+}
+
+#[cargo_test]
+fn rustc_caching_allow_first() {
+    let p = make_project_with_rustc_warning();
+    p.cargo("check")
+        .masquerade_as_nightly_cargo(&["warnings"])
+        .arg("-Zwarnings")
+        .arg("--config")
+        .arg("build.warnings='allow'")
+        .with_stderr_data(ALLOW_CLEAN.clone())
```

Comment:
We should probably have the actual output here, so you don't have to jump back and forth when figuring out what a test is doing.

Reply:
This change was made in the feature commit, rather than the test commit, making it harder to see what the feature commit did

Reply:
Thanks